java/catalog-adapter-hms/src/main/java/com/lancedb/catalog/adapter/hms/service/HmsCatalogService.java

```diff
@@ -0,0 +1,333 @@
+      String location,
+      Map<String, String> properties) {
+
+    return buildTable(dbName, tableName, schema)
+        .withLocation(location)
+        .withProperties(properties)
+        .create();
+  }
+
+  private Table loadHmsTable(String dbName, String tableName)
+      throws TException, InterruptedException {
+    try {
+      return clients.run(client -> client.getTable(dbName, tableName));
+    } catch (NoSuchObjectException nte) {
+      LOGGER.trace("Table not found {}", dbName + "." + tableName, nte);
+      return null;
+    }
+  }
+
+  public HMSTableBuilder buildTable(String dbName, String tableName, Schema schema) {
+    return new HMSTableBuilder(dbName, tableName, schema);
+  }
+
+  protected class HMSTableBuilder {
+
+    private final String dbName;
+    private final String tableName;
+    private final Schema schema;
+    private String location = null;
+    private Map<String, String> storageOption;
+
+    public HMSTableBuilder(String dbName, String tableName, Schema schema) {
+      Preconditions.checkArgument(
+          StringUtils.isNotBlank(dbName), "Database can not be null or empty.");
+      Preconditions.checkArgument(
+          StringUtils.isNotBlank(tableName), "Table can not be null or empty.");
+
+      this.dbName = dbName;
+      this.tableName = tableName;
+      this.schema = schema;
+    }
+
+    public HMSTableBuilder withLocation(String location) {
+      this.location = location;
+      return this;
+    }
+
+    public HMSTableBuilder withProperties(Map<String, String> properties) {
+      this.storageOption = properties;
+      return this;
+    }
+
+    public HMSTableBuilder withProperty(String key, String value) {
+      this.storageOption.put(key, value);
+      return this;
+    }
+
+    public Dataset create() {
+      try (RootAllocator allocator = new RootAllocator(Long.MAX_VALUE)) {
+        Table hmsTable = newHmsTable("admin", dbName, tableName, null);
+
+        LOGGER.info("Creating table: {}.{} in hms.", dbName, tableName);
+        clients.run(
+            client -> {
+              client.createTable(hmsTable);
+              return null;
+            });
+
+        Table hmsTbl = loadHmsTable(dbName, tableName);
+        if (hmsTbl != null) {
+          String dsLocation = Utils.renameScheme(hmsTbl.getSd().getLocation());
+          LOGGER.info("Creating dataset: {}.{} under {}", dbName, tableName, location);
+
+          return Dataset.create(
+              allocator,
+              dsLocation,
+              schema,
+              new WriteParams.Builder().withStorageOptions(this.storageOption).build());
+        } else {
+          throw new RuntimeException(
+              "Failed to create dataset (can not load): " + dbName + "." + tableName);
+        }
+      } catch (TException e) {
+        throw new RuntimeException(e);
+      } catch (InterruptedException e) {
+        throw new RuntimeException(e);
+      }
+    }
+
+    public Dataset register() {
```

Comment:
looks like we don't really have a registerTable API yet. Do you plan to introduce one?

Reply:
Yes, we already have. I just have not exposed it right now.